### PR TITLE
Parallel e2e tests

### PR DESCRIPTION
**Pull request by mmatczuk**
This patchset redefines our e2e tests framework allowing to run the tests in parallel.
It also allows to run remotely with `podman-remote` as the container runtime.

Check [e2e/README.md](https://github.com/saucelabs/forwarder/blob/mmt/parallel_e2e/e2e/README.md) for details.

### CODE REVIEW

File: e2e/setup/runner.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+	g, _ := errgroup.WithContext(ctx)
+	if r.Parallel > 0 {
+		g.SetLimit(r.Parallel)
+	}
+	if r.Debug {
+		g.SetLimit(1)
+	}
+
+	setups := slices.Clone(r.Setups)
+	if r.Parallel != 1 {
```
Comment (Choraden) [inline e2e/setup/runner.go:46]: nit: `r.Parallel != 1 && !r.Debug`

**Comment by Choraden** [comment]
LGTM, the debug mode satisfies my needs. 

I'm a little surprised the e2e job takes 5m. I thought we would be able to reduce it even more, but it is still a great improvement as non parallel takes 8m on average.

**Comment by mmatczuk** [comment]
Adding more parallelism does not make it run faster due to constrained resources.
We could run it in two shards odd and even to make it twice as fast.

**Comment by mmatczuk** [comment]
https://github.com/saucelabs/forwarder/issues/770

**Pull request merged**